src/disk_interface_test.cc

```diff
@@ -135,9 +135,11 @@ TEST_F(DiskInterfaceTest, StatCache) {
   EXPECT_EQ(disk_.Stat("subdir", &err),
             disk_.Stat("subdir/.", &err));
   EXPECT_EQ("", err);
+  /* This is not necessarily true, not even uncached.
   EXPECT_EQ(disk_.Stat("subdir", &err),
```

Comment:
Why L99 is OK, but here is not?

Reply:
I didn't see that first. It seems to be different behaviour of FindFirstFileExA() in the cached behaviour and GetFileAttributesEx() in the non-cached flavour. Have no time to investigate it further today. Does anyone else know any details or have a suggestion how to fix it?

Reply:
Is it better to do path canonicalization around
https://github.com/ninja-build/ninja/blob/e234a7bdb6c42f4539c0ab09b624f191287c2c10/src/disk_interface.cc#L165 ?